### PR TITLE
Add PLUTO + Changefile metadata

### PR DESCRIPTION
**Pull request by alexrichey**
Adds PLUTO + Changefile metadata, and adds various tweaks / fixes to make the push process more ergonomic. 

The commits are atomic, so you can step through them to get a sense of the changes. E.g. I generated the skeleton of the PLUTO metadata using our automated tool, and then filled in the details in a separate commit. 

Regarding the actual PLUTO metadata, I wouldn't worry too much about reviewing all 2k lines. We'll need to do a review with GIS at some point in the future. But it'll be a huge win to have one source of truth for all that metadata.

### CODE REVIEW

File: dcpy/connectors/socrata/metadata.py [hunk 1]
```diff
@@ -77,7 +78,7 @@ def make_dcp_metadata(socrata_md: pub.Socrata.Responses.Metadata) -> models.Meta
         summary=socrata_md["description"],
         description=socrata_md["description"],
         tags=socrata_md["tags"],
-        each_row_is_a=socrata_md["metadata"]["rowLabel"],
+        each_row_is_a=socrata_md["metadata"].get("rowLabel") or "<FILL_ME_IN>",
```
Comment (sf-dcp) [inline dcpy/connectors/socrata/metadata.py:81]: `<FILL_ME_IN>` <-- small nit: perhaps make it a constant to be used across the file?

File: products/pluto/metadata.yml [hunk 2]
```diff
@@ -0,0 +1,675 @@
+name: primary_land_use_tax_lot_output_pluto
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:1]: curious, is this the unique id in Socrata? If I understand correctly, command to generate metadata is 
```bash
python3 metadata.py export <four_four> <output_path>
```

File: products/pluto/metadata.yml [hunk 3]
```diff
@@ -1,675 +1,1705 @@
...
-  separated values (CSV) file format. The PLUTO files contain more than seventy fields\
-  \ derived from data maintained by city agencies.\r\n\r\nAll previously released\
-  \ versions of this data are available at <a href=\"https://www1.nyc.gov/site/planning/data-maps/open-data/bytes-archive.page?sorts[year]=0\"\
-  >BYTES of the BIG APPLE- Archive</a>"
-description: "Extensive land use and geographic data at the tax lot level in comma\u2013\
-  separated values (CSV) file format. The PLUTO files contain more than seventy fields\
-  \ derived from data maintained by city agencies.\r\n\r\nAll previously released\
-  \ versions of this data are available at <a href=\"https://www1.nyc.gov/site/planning/data-maps/open-data/bytes-archive.page?sorts[year]=0\"\
-  >BYTES of the BIG APPLE- Archive</a>"
+summary: ""
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:3]: what's the difference between summary and description tags?

File: products/pluto/metadata.yml [hunk 4]
```diff
@@ -1,675 +1,1705 @@
...
-each_row_is_a: <FILL_ME_IN>
+each_row_is_a: Tax Lot
+
 destinations:
-- id: socrata_prod
+- id: socrata_prod_water_incl
   type: socrata
   four_four: 64uk-42ks
-  attachments: []
+  attachments:
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:18]: I noticed `attachments` tag is here and on the line 43. Can they have different attachments?

File: products/pluto/metadata.yml [hunk 5]
```diff
@@ -1,675 +1,1705 @@
...
+
+      Two portions of the city, Marble Hill and Rikers Island, are legally located in one
+      borough but are serviced by a different borough. The BOROUGH codes associated
+      with these areas are the boroughs in which they are legally located.
+      Marble Hill is serviced by the Bronx, but is legally located in Manhattan and has a
+      BOROUGH of MN. Rikers Island is serviced by Queens, but is legally located in the
+      Bronx and has a BOROUGH of BX.
+    data_type: text
+    readme_data_type: Alphanumeric - 2 characters
+    data_source: |
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:277]: This is so cool! Love the level of detail per column

File: products/pluto/metadata.yml [hunk 6]
```diff
@@ -1,675 +1,1705 @@
...
+
+  - name: block
+    display_name: Tax block
+    description: |
+      The tax block in which the tax lot is located.
+      This field contains a one to five-digit tax block number.
+      Each tax block is unique within a borough (see BOROUGH).
+    data_type: integer
+    data_source: Department of Finance - Property Tax System (PTS)
+    readme_data_type: Numeric - 5 digits (99999)
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:296]: Q about `readme_data_type`: does this mean a value is up to 5 digits, not exactly 5 digits?

File: products/pluto/metadata.yml [hunk 7]
```diff
@@ -1,675 +1,1705 @@
...
+    description: |
+      The name of the owner of the tax lot.
+
+      For publicly owned tax lots, owner names have been normalized. For example, “NYC PARKS”, “PARKS DEPARTMENT”, and “PARKS AND RECREATION (GENERAL)” have been changed to “NYC DEPARTMENT OF PARKS AND RECREATION”.
+
+      If OWNER NAME is normalized, DCPEdited is set to “1”. (see CHANGED BY DCP).
+    data_source: |
+      Department of Finance - Property Tax System (PTS)
+      Department of City Planning – [PLUTO_input_research.csv](https://github.com/NYCPlanning/db-pluto/blob/dev/pluto_build/data/pluto_input_research.csv), field ownername
+    # TODO: should we have elements like links as markdown, or html?
```
Comment (sf-dcp) [inline products/pluto/metadata.yml:867]: Probably depends if we can transfer the markdown format to the generated pluto metadata? i.e. link name + link

File: products/pluto/metadata_change_file.yml
Comment (sf-dcp) [inline products/pluto/metadata_change_file.yml:1]: is this file generated programmatically or manually?

File: dcpy/connectors/socrata/publish.py [hunk 8]
```diff
@@ -357,27 +356,25 @@ def make_socrata_metadata(dcp_md: models.Metadata):
...
-    shape_file_path = (
-        dataset_package_path / "dataset_files" / shapefile_name
+    md_dataset = metadata.package.get_dataset(ds_name_to_push)
+    file_path = (
+        dataset_package_path / "dataset_files" / md_dataset.filename
     )  # TODO: this isn't the right place for this calculation. Move to lifecycle.package.
 
     dataset = Dataset(four_four=dest.four_four)
 
-    dataset.discard_open_revisions()
```
Comment (fvankrieken) [inline dcpy/connectors/socrata/publish.py:358]: What had been the reason for this before, and why the change?

Comment (fvankrieken) [reply dcpy/connectors/socrata/publish.py:358]: I see that it was just moved down :)

Still curious as to reason!

**Review by fvankrieken — APPROVED** [review]
Good stuff. Our march to metadata sanity continues!

File: products/pluto/metadata.yml (hunk 4 above)
Comment (damonmcc) [reply products/pluto/metadata.yml:18]: I think they can, especially when we expand the destinations beyond just `socrata`

File: products/pluto/metadata.yml (hunk 7 above)
Comment (damonmcc) [reply products/pluto/metadata.yml:867]: markdown could be nice, but html might be more compatible for pushing straight to destinations and for generating docs

**Review by damonmcc — APPROVED** [review]

**Comment by alexrichey** [comment]
@fvankrieken and @sf-dcp gonna merge this (for @damonmcc 's sake) and respond to comments, fyi

File: products/pluto/metadata_change_file.yml
Comment (alexrichey) [reply products/pluto/metadata_change_file.yml:1]: @sf-dcp These files are generally a mix of programmatic generation, then manual fixups. I added a Socrata connector to export the metadata from a Socrata page, e.g.
```
python -m dcpy.cli connectors socrata metadata export v2h8-6mxf
```
but those generally have partial information (e.g. they're missing column descriptions), and the formatting tends to be wonky, so they require some fixup

File: products/pluto/metadata.yml (hunk 7 above)
Comment (alexrichey) [reply products/pluto/metadata.yml:867]: yeah, very much an open question. Happy to hear thoughts 🙂

File: products/pluto/metadata.yml (hunk 6 above)
Comment (alexrichey) [reply products/pluto/metadata.yml:296]: I think it means exactly, but I'm not sure. The intent for this field is just to generate the readme - it's what we'll display as the type rather than our internal type - so it's copied straight from the PLUTO readme.

File: dcpy/connectors/socrata/metadata.py (hunk 1 above)
Comment (alexrichey) [reply dcpy/connectors/socrata/metadata.py:81]: That's a great idea. We could also reuse that variable to raise an error when we deserialize the metadata and we see that it contains our `<FILL_ME_IN>` placeholder.

File: products/pluto/metadata.yml (hunk 6 above)
Comment (alexrichey) [reply products/pluto/metadata.yml:296]: this also makes me start to think about documenting our Python models, so that it's obvious what a field like this is for. Looks like there are some neat libraries like [autodoc-pydantic](https://autodoc-pydantic.readthedocs.io/en/stable/users/examples.html)
